core/cmd_exec.py

```diff
@@ -275,13 +275,13 @@ def _RunBuiltin(self, builtin_id, argv):
       status = builtin.Jobs(argv, self.job_state)
 
     elif builtin_id == EBuiltin.PUSHD:
-      status = builtin.Pushd(argv, self.dir_stack)
+      status = builtin.Pushd(argv, self.mem, self.dir_stack)
```

Comment:
Although it's a little redundant, I think all three functions should just take `self.mem.GetVar('HOME')`

In general it's better to have a narrower interface, and not depend on all of `Mem()` when you only need to read a single variable.

Reply:
True, it does only need one variable. I can change it to just take the value of $HOME specifically